integration-tests/tests/client/process_blocks.rs

```diff
@@ -1917,10 +1918,27 @@ fn test_block_merkle_proof_with_len(n: NumBlocks) {
     let mut env = TestEnv::new(ChainGenesis::test(), 1, 1);
     let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap().clone();
     let mut blocks = vec![genesis_block.clone()];
-    for i in 1..n {
-        let block = env.clients[0].produce_block(i).unwrap().unwrap();
-        blocks.push(block.clone());
-        env.process_block(0, block, Provenance::PRODUCED);
+    let mut rng = rand::thread_rng();
+    let mut cur_height = genesis_block.header().height() + 1;
+    while cur_height < n {
+        let should_fork = rng.gen_bool(0.5);
```

Comment:
FWIW, I am not a fan of "just pull `thread_rng()` out of thin air when you need" for large projects. I usually prefer to explicitly thread rng as a parameter. This helps when you figure out that the stuff does break, and you want to fix a seed to debug it. 

As Longarithm recently pointed out to me, `rand::thread_rng` doesn't have reseeding API by design. So debugging tests becomes more cumbersome, as you first need to switch the rng. 

But, that's a minor point, not that it make big practical difference for a single case!